onnx/defs/data_type_utils.cc

```diff
@@ -243,13 +255,20 @@ bool DataTypeUtils::IsValidDataTypeString(const std::string& type_str) {
   return (allowedSet.find(type_str) != allowedSet.end());
 }
 
-void DataTypeUtils::FromDataTypeString(
+Common::Status DataTypeUtils::FromDataTypeString(
     const std::string& type_str,
     int32_t& tensor_data_type) {
-  assert(IsValidDataTypeString(type_str));
+  if (!IsValidDataTypeString(type_str))
+    return Common::Status(
+        Common::NONE, Common::INVALID_ARGUMENT, "invalid type string");
 
   TypesWrapper& t = TypesWrapper::GetTypesWrapper();
-  tensor_data_type = t.TypeStrToTensorDataType()[type_str];
+  auto iter = t.TypeStrToTensorDataType().find(type_str);
+  if (iter == t.TypeStrToTensorDataType().end())
+    return Common::Status(
+        Common::NONE, Common::INVALID_ARGUMENT, "invalid type string");
```

Comment:
```suggestion
        Common::NONE, Common::INVALID_ARGUMENT, "invalid type string: ", type_str);
```
At least user can search for the returned `type_str` in their code to find the error.